### ISSUE TITLE
Error: free(): double free detected in tcache 2

### ISSUE THREAD

**Issue by S1LV3RJ1NX** [fdabrandao/vpsolver#8]
Hello,
I am getting this wierd error stating:
```
>>> vbpsol...
VPSolver 3.1.3, Copyright (C) 2013-2016, Filipe Brandao
free(): double free detected in tcache 2
/data/prathamesh_home/bin_packing_heuristics/venv/bin/vpsolver_scip.sh: line 186: 2406799 Aborted                 
(core dumped) vbpsol $afg_file $TMP_DIR/vars.sol $vbpsol_opts
```
Even after optimizers like GUROBI or SCIP state optimal solution found, during vbpsol phase I get this error. 
```
SCIP Status        : problem is solved [optimal solution found]
Solving Time (sec) : 148.32
Solving Nodes      : 1
Primal Bound       : +3.00000000000000e+00 (1 solutions)
Dual Bound         : +3.00000000000000e+00
Gap                : 0.00 %
```
How to fix?

**Comment by fdabrandao** [comment]
@S1LV3RJ1NX, could you please provide the instance that results in this error?

**Comment by S1LV3RJ1NX** [comment]
@fdabrandao I am using this in my simulator where the requests are converted to MVP format at runtime. Hence it would be difficult to share the instance. The scenario is I take a batch of requests and fit it into bins, then whatever bin sizes are remaining, I use the next batch of requests to fit into these newly sized bins. It satisfies a certain number of requests, but after some time, it fails and throws the above error. 

Things I have checked:
- All the requests are within the bounds and are satisfiable.

What is the meaning of this error ?

**Comment by fdabrandao** [comment]
Can you modify the application to display the MVP file before solving so that you can see the one that resulted in the error? This error means that something went wrong while retrieving the solution from the arc-flow graph. Having the MVP instance that caused this would be really important to know what went wrong.

**Comment by S1LV3RJ1NX** [comment]
Yaa, I totally understand. Hence, I have quickly made a subset of my program that will help you run this. It gives the same error. Please have a look at it.
[binpacking.zip](https://github.com/fdabrandao/vpsolver/files/9290007/binpacking.zip)

**Comment by S1LV3RJ1NX** [comment]
Hello @fdabrandao, did you find the issue?

**Comment by fdabrandao** [comment]
@S1LV3RJ1NX, I still haven't found the exact same issue you had yet but I got a different error. One thing I noticed is that you use many bin types for bins of the same size. It makes more sense to group bin types that share the same size and cost.

**Comment by S1LV3RJ1NX** [comment]
@fdabrandao What you have said is correct. But I need to know precisely in which bin it went, as I have bin names for each bin, so once that bin has some allocation, I reduce its size and use it as a new bin later on. How do I know to which bin it was allocated if I combine bins with similar parameters?

**Comment by S1LV3RJ1NX** [comment]
@fdabrandao I got the exact bins following your code. After sometime I am getting following assertion error:
`AssertionError: assertion `Ws[t][d] >= 0` failed in "src/instance.cpp" line 144` What does this mean? It is because bin capacities should be more than 0 and 0 are not allowed?

**Comment by fdabrandao** [comment]
@S1LV3RJ1NX, yes, if you get  the assertion error for nfits >= 0 it means that some item could not be allocated anywhere and the problem is known to be infeasible immediately.
